AWSKinesis/AWSAbstractKinesisRecorder.m

```diff
@@ -178,17 +178,14 @@ - (AWSTask *)saveRecord:(NSData *)data
                                ];
                 if (!result) {
                     AWSLogError(@"SQLite error. [%@]", db.lastError);
-                    error = db.lastError;
+                    dbError = db.lastError;
                 }
             }];
         }
-
-        if (error) {
```

Comment:
Since `AWSTask` can expose only one error message, whenever we encounter an issue, we should return it right away. Otherwise, the SDK potentially encounters multiple errors, but returns only one of them. (There are ways to wrap around an error inside of an error, but it's not a common practice in Cocoa Framework.)